mypy/checker.py

```diff
@@ -2553,6 +2553,8 @@ def check_multi_assignment(self, lvalues: List[Lvalue],
         elif isinstance(rvalue_type, UnionType):
             self.check_multi_assignment_from_union(lvalues, rvalue, rvalue_type, context,
                                                    infer_lvalue_type)
+        elif rvalue_type.type.fullname == 'builtins.str':
```

Comment:
You need to protect this with an `isinstance` check.

Reply:
Yeah it was my original thought to match on `isinstance(rvalue_type, str)` but then this conditional branch will not be triggered even though `rvalue_type.type.fullname == 'builtins.str' ` does get triggered.

Reply:
Protected with an `isinstance(rvalue_type, Instance)` check, probably. Note that checking mypy with itself produces a type error now.

Reply:
I see thanks, I made the change, it made most of the tests pass for me, except a few cases where the actual output of a test included the error message which made the tests fail due to a mismatch.